server/RevShop/src/test/java/net/revature/RevShop/UserTesting/UserRepositoryTest.java

```diff
@@ -0,0 +1,88 @@
+package net.revature.RevShop.UserTesting;
+
+import net.revature.RevShop.Models.User;
+import net.revature.RevShop.Repositories.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@SpringBootTest
+public class UserRepositoryTest {
+    //mock prep mock , inject mock and before each
+    @Mock
+    private UserRepository userRepository;
+    private User classUndertest;
+    Long MyIdL = 1L;
+
+    @BeforeEach
+    public void setUp() {
+        classUndertest = new User(
+                "YuQi",
+                "Yuqi@Gidle.com",
+                "Freak",
+                "Yuqi",
+                "Song",
+                User.UserType.BUYER,
+                "Test Business",
+                false,
+                LocalDateTime.now()
+        );
+    }
+
+    //All testing
+    @Test
+    public void testingSaveUser() {
```

Comment:
Testing the Repository is not necessary since it is implemented by an external library Spring Data JPA Hibernate. You want to only test code that is being written by yourself and teammates.

Reply:
The Repository library which is Hibernate is used by millions of devs so at this point we know it works. Thats why we don't test external libraries.